src/sources.js

```diff
@@ -286,6 +286,62 @@ class CommunityNarrative extends Narrative {
   }
 }
 
+
+class UrlDefinedSource extends Source {
+
+  static get _name() { return "urlDefined"; }
+  get baseUrl() {
+    throw new Error("UrlDefinedSource does not use `this.baseUrl`");
+  }
+
+  dataset(pathParts) {
+    return new UrlDefinedDataset(this, pathParts);
+  }
+  narrative(pathParts) {
+    return new UrlDefinedNarrative(this, pathParts);
+  }
+
+  // what does this mean in the context of this source?
+  async availableDatasets() { return []; }
```

Comment:
These three methods aren't applicable (but are still run) for this Source. I.e. the `getAvailable` request is still made, even though there is no relevant data to be provided.

Reply:
94f91a5 tidies up the comments here a bit